src/Core/Services/IReferenceEventService.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System.Threading.Tasks;
+using Bit.Core.Models;
+
+namespace Bit.Core.Services
+{
+    public interface IReferenceEventService
+    {
+        Task RaiseEventAsync(IReferenceable reference, string eventType,
+            object additionalInfo = null);
```

Comment:
Is it possible to structure this a bit more? Maybe a <string, object> dictionary?

Reply:
The goal is raw JSON serialization and platform agnostic deserialization. I could, however, just define a class with all of the possible properties we need to capture so it's more structured, less random. I think that would be more straightforward than a dictionary (which would just add more syntactical overhead/curly-brace madness).

Reply:
probably fine as-is